### PR TITLE
chore: rename j_must_have to j_deprecated and only warn about deprecated keys

### LINKED ISSUE #3523: rename j_must_have to j_deprecated and only warn about deprecated keys
maybe rename this to j_deprecated and only warn about deprecated keys,
if the deprecated_key argument is left empty function puppose is only custom
error since dict[key] already raises KeyError when the key is missing

Line: 184
https://github.com/deepmodeling/deepmd-kit/blob/76371ceed2fd0d8e259e482e046da350d4006c0a//deepmd/common.py#L181-L191

### PR DESCRIPTION
**Pull request by njzjz**
Fix #3523.

## Summary by CodeRabbit

- **Refactor**
  - Simplified code by removing the `j_must_have` function and directly accessing dictionary keys in various test files.
  - Replaced `j_must_have` with direct dictionary access for improved code readability and maintenance.

- **Chores**
  - Updated test files to directly access dictionary values, enhancing code readability and maintainability.

### CODE REVIEW

File: deepmd/utils/compat.py [hunk 1]
```diff
@@ -378,7 +378,7 @@ def is_deepmd_v0_input(jdata):
         return "model" not in jdata.keys()
 
     def is_deepmd_v1_input(jdata):
-        return "systems" in j_must_have(jdata, "training").keys()
+        return "systems" in jdata, ["training"].keys()
```
Comment (njzjz) [inline deepmd/utils/compat.py:381]:
```suggestion
        return "systems" in jdata["training"].keys()
```

File: source/tests/tf/test_descrpt_hybrid.py [hunk 2]
```diff
@@ -40,10 +37,10 @@ def test_descriptor_hybrid(self):
         jfile = "water_hybrid.json"
         jdata = j_loader(jfile)
 
-        systems = j_must_have(jdata, "systems")
+        systems = jdata["systems"]
         set_pfx = "set"
-        batch_size = j_must_have(jdata, "batch_size")
-        test_size = j_must_have(jdata, "numb_test")
+        batch_size = jdata["batch_size"]
```
Comment (njzjz) [inline source/tests/tf/test_descrpt_hybrid.py:42]:
```suggestion
```

File: source/tests/tf/test_descrpt_hybrid.py [hunk 3]
```diff
@@ -40,10 +37,10 @@ def test_descriptor_hybrid(self):
...
+        test_size = jdata["numb_test"]
```
Comment (njzjz) [inline source/tests/tf/test_descrpt_hybrid.py:43]:
```suggestion
```

File: source/tests/tf/test_descrpt_se_a_type.py [hunk 4]
```diff
@@ -33,15 +30,15 @@ def test_descriptor_two_sides(self):
         jfile = "water_se_a_type.json"
         jdata = j_loader(jfile)
 
-        systems = j_must_have(jdata, "systems")
+        systems = jdata["systems"]
         set_pfx = "set"
-        batch_size = j_must_have(jdata, "batch_size")
-        test_size = j_must_have(jdata, "numb_test")
+        batch_size = jdata["batch_size"]
+        test_size = jdata["numb_test"]
```
Comment (njzjz) [inline source/tests/tf/test_descrpt_se_a_type.py:36]:
```suggestion
```

File: source/tests/tf/test_polar_se_a_tebd.py [hunk 5]
```diff
@@ -48,14 +45,14 @@ def test_model(self):
         jfile = "polar_se_a_tebd.json"
         jdata = j_loader(jfile)
 
-        systems = j_must_have(jdata, "systems")
+        systems = jdata["systems"]
         set_pfx = "set"
-        batch_size = j_must_have(jdata, "batch_size")
-        test_size = j_must_have(jdata, "numb_test")
+        batch_size = jdata["batch_size"]
```
Comment (njzjz) [inline source/tests/tf/test_polar_se_a_tebd.py:50]:
```suggestion
```

File: source/tests/tf/test_polar_se_a.py [hunk 6]
```diff
@@ -45,14 +42,14 @@ def test_model(self):
         jfile = "polar_se_a.json"
         jdata = j_loader(jfile)
 
-        systems = j_must_have(jdata, "systems")
+        systems = jdata["systems"]
         set_pfx = "set"
-        batch_size = j_must_have(jdata, "batch_size")
-        test_size = j_must_have(jdata, "numb_test")
+        batch_size = jdata["batch_size"]
+        test_size = jdata["numb_test"]
```
Comment (njzjz) [inline source/tests/tf/test_polar_se_a.py:48]:
```suggestion
```

File: source/tests/tf/test_model_spin.py [hunk 7]
```diff
@@ -48,18 +45,18 @@ def test_model_spin(self):
 
         # set system information
         set_pfx = "set"
-        batch_size = j_must_have(jdata["training"]["training_data"], "batch_size")
+        batch_size = jdata["training"]["training_data"]["batch_size"]
```
Comment (njzjz) [inline source/tests/tf/test_model_spin.py:48]:
```suggestion
```

File: source/tests/tf/test_model_se_t.py [hunk 8]
```diff
@@ -36,14 +33,14 @@ def test_model(self):
         jfile = "water_se_t.json"
         jdata = j_loader(jfile)
 
-        systems = j_must_have(jdata, "systems")
+        systems = jdata["systems"]
         set_pfx = "set"
-        batch_size = j_must_have(jdata, "batch_size")
-        test_size = j_must_have(jdata, "numb_test")
+        batch_size = jdata["batch_size"]
+        test_size = jdata["numb_test"]
```
Comment (njzjz) [inline source/tests/tf/test_model_se_t.py:39]:
```suggestion
```

File: source/tests/tf/test_descrpt_se_a_type.py [hunk 9]
```diff
@@ -197,15 +191,15 @@
...
```
[the remark this answers is not in the document]
Comment (njzjz) [reply source/tests/tf/test_descrpt_se_a_type.py:197]: @coderabbitai resolve

**Comment by njzjz** [comment]
https://github.com/coderabbitai resolve

**Comment by njzjz** [comment]
@coderabbitai resolve

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**